exercises-cpp/mohammed_almakki/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,123 @@
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <regex>
+#include <unordered_map>
+#include <cstring>
+#include <iomanip>
+
+std::string readFile(const std::string &filepath)
+{
+    std::ifstream inputStream(filepath);
+
+    if (inputStream)
+    {
+        std::ostringstream ss;
+        ss << inputStream.rdbuf();
+        return ss.str();
+    }
+    else
+    {
+        throw std::invalid_argument("Invalid path. Please make sure the file path is correct and the file exists.");
+    }
+}
+
+std::vector<std::string> tokenizeFile(const std::string &file)
+{
+    std::regex re("[-\\s]");
```

Comment:
This regex is used to split the string by two delimiters, the space, and the hyphen. It will match a space or a hyphen. After that, the expression is used in a reverse way, to get all the words that don't match it i.e. the split words.

Reply:
A Hyphen is also a special character and will need to be exited as well

Reply:
Yes, it will be removed. For example, the string "lines are hard-coded" will be transformed into a vector of ["lines", "are", "hard", "coded"] by this expression.